### ISSUE TITLE
RNG seeding for Monte Carlo lab

### ISSUE THREAD

**Issue by erose** [grusky/cs150labs#10]
http://stackoverflow.com/questions/11526975/set-random-seed-programwide-in-python

Apparently if you set `random`'s seed in the main module's import, it holds for all other imports of `random` as well. So if there was a main module that called the submitted program, we could make a test for Monte Carlo.

It seems like the submitted are being run from a shell script though, not through a main module -- is that accurate?

**Comment by grusky** [comment]
We don't need a main method if we do a little bit of hacky importing. Try this:

```python
# monte.py

import random
import math

print("This program calculates the value of Pi by simulating the throwing of darts onto a round target on a square background.")

darts = int(input("How many darts do you want to throw? "))

hits = 0
for i in range(darts):
    x = random.uniform(-1.0, 1.0)
    y = random.uniform(-1.0, 1.0)

    if math.sqrt( x*x + y*y ) <= 1:
        hits += 1

pi = 4 * hits/darts
error = round(abs(pi - math.pi) / math.pi * 100, 2)

print("The value of Pi after ", darts, " iterations is ", pi ," which is off by ", error,"%.", sep = "")
```

```python
# monteseed.py

import random
random.seed(5)
import monte
```

This is why I think the grading scripts really should be written in Python. The downside is that the script must be in the same directory as the student's code. But this can be fixed!

```python
# monteseed.py

import sys
import random

random.seed(5)

sys.path.append(os.path.abspath(sys.argv[1]))
import monte
```

And that's easily usable with whatever grading system they end up going with:

```bash
echo 100 | python3 monteseed.py lab03/student
```

```
This program calculates the value of Pi by simulating the throwing of darts onto a round target on a square background.
How many darts do you want to throw? The value of Pi after 100 iterations is 2.96 which is off by 5.78%.
```

And if you run the same code on your computer, you should get exactly the same result! Unless, of course, they seed random with their own number at the start of their code. But then they deserve to get the wrong answer because it really isn't random, is it?

I'm guessing this kind of thing can replace main methods most of the time.

**Comment by erose** [comment]
Yeah, that sounds like a good way to go. I was actually talking about a main *module*, not method. By that I just meant a separate module that calls the submitted code, which is what you're talking about too.

We might be able to avoid even writing monteseed.py if we use python's -c command line argument. That lets you supply a string which gets evaluated in the Python shell.

Here's our file:

````
# rand.py

import random
print(random.random())
````
and the output:

````
eli@tintagel:~$ python -c "import random; random.seed(1); import rand"
0.134364244112
eli@tintagel:~$ python -c "import random; random.seed(1); import rand"
0.134364244112
eli@tintagel:~$ python -c "import random; random.seed(1); import rand"
0.134364244112
````

So in general, executing file `a.py` with `python a.py` could be replaced by running `python -c "import a"`. This seems like it might screw something up -- can anyone think of what?

**Comment by erose** [comment]
Okay, yeah, how wou
